python/lsst/meas/extensions/multiprofit/plots.py

```diff
@@ -0,0 +1,361 @@
+
+    @abstractmethod
+    def get_flux(self, band: str) -> np.ndarray:
+        """Return the flux in a given band."""
+
+    @abstractmethod
+    def get_id(self) -> np.ndarray:
+        """Return a unique source id."""
+
+    @abstractmethod
+    def get_is_extended(self) -> np.ndarray:
+        """Return if the source is extended."""
+
+    @abstractmethod
+    def get_is_variable(self) -> np.ndarray:
+        """Return if the source is variable."""
+
+    @abstractmethod
+    def get_x(self) -> np.ndarray:
+        """Return the x pixel coordinates."""
+
+    @abstractmethod
+    def get_y(self) -> np.ndarray:
+        """Return the y pixel coordinates."""
+
+
+@dataclass(frozen=True, config=pydantic.ConfigDict(arbitrary_types_allowed=True))
+class TruthSummaryTable(ObjectTableBase):
+    """Class for retrieving columns from DC2 truth tables."""
+
+    def get_flux(self, band: str) -> np.ndarray:
+        return self.table[f"flux_{band}"]
+
+    def get_id(self) -> np.ndarray:
+        return self.table["id"]
+
+    def get_is_extended(self) -> np.ndarray:
+        return self.table["is_pointsource"] == False  # noqa: E712
+
+    def get_is_variable(self) -> np.ndarray:
+        return self.table["is_variable"] == True  # noqa: E712
+
+    def get_x(self):
+        return self.table["x"]
+
+    def get_y(self):
+        return self.table["y"]
+
+
+@dataclass(frozen=True, config=pydantic.ConfigDict(arbitrary_types_allowed=True))
+class ObjectTable(ObjectTableBase, ABC):
+    """Base class for objectTable_tract."""
+
+    def get_id(self) -> np.ndarray:
+        return self.table["objectId"]
+
+    def get_is_extended(self) -> np.ndarray:
+        return self.table["refExtendedness"] >= 0.5
+
+    def get_is_variable(self) -> np.ndarray:
+        return np.zeros(len(self.table), dtype=bool)
+
+    def get_x(self):
+        return self.table["x"]
+
+    def get_y(self):
+        return self.table["y"]
+
+
+@dataclass(frozen=True, config=pydantic.ConfigDict(arbitrary_types_allowed=True))
+class ObjectTableCModel(ObjectTable):
+    """Class for retrieving CModel fluxes from objectTable_tract."""
+
+    def get_flux(self, band: str) -> np.ndarray:
+        return self.table[f"{band}_cModelFlux"]
+
+
+@dataclass(frozen=True, config=pydantic.ConfigDict(arbitrary_types_allowed=True))
+class ObjectTableMultiProFit(ObjectTableBase):
+    """Class for retrieving fluxes from objectTable_tract_multiprofit."""
+
+    name_model: str = pydantic.Field(doc="The name of the MultiProFit model")
+    prefix_col: str = pydantic.Field(doc="The prefix for object fit columns", default="mpf_")
+
+    def get_flux(self, band: str) -> np.ndarray:
+        return self.table[f"{self.prefix_col}{self.name_model}_{band}_flux"]
+
+    def get_id(self) -> np.ndarray:
+        return self.table["objectId"]
+
+    def get_is_extended(self) -> np.ndarray:
+        return self.table["refExtendedness"] >= 0.5
```

Comment:
No, it's propagated from the `objectTable_tract`. If you mean why use that column instead of `refSizeExtendedness`, I just haven't updated the downstream tasks to do that yet.